operators/pkg/instctrl/cloudinit.go

```diff
@@ -86,6 +87,33 @@ func (r *InstanceReconciler) EnforceCloudInitSecret(ctx context.Context) error {
 	return nil
 }
 
+// Get the specs of the volume
+func (r *InstanceReconciler) GetNFSSpecs(ctx context.Context) (serverName string, path string, err error) {
+	instance := clctx.InstanceFrom(ctx)
+	secretName := types.NamespacedName{Namespace: instance.Namespace, Name: NFSSecretName}
+
+	secret := corev1.Secret{}
+	if err = r.Get(ctx, secretName, &secret); err != nil {
+		ctrl.LoggerFrom(ctx).Error(err, "failed to retrieve secret", "secret", secretName)
+		return
+	}
+
+	if serverNameBytes, ok := secret.Data[NFSSecretServerNameKey]; !ok {
+		err = fmt.Errorf("cannot find %v key in secret", NFSSecretServerNameKey)
+		ctrl.LoggerFrom(ctx).Error(err, "failed to retrieve NFS spec from secret", "secret", secretName)
+		return
+	} else {
```

Comment:
The else block is not required, since you previously returned in case of error.

Reply:
Removed else block